src/application/communication/GameClient.java

```diff
@@ -163,7 +185,13 @@ private void joinGame(Pane gameBox, Specializations spec)
         System.out.println("trying to join game");
         try
         {
-            Broker broker = new Broker();
+            if(broker != null)
+            {
+                broker.unregisterAll(this);
```

Comment:
i see your point. however i think gameclient should stop observing broker when it deactivates. so maybe that functionality must be called from broker